compiler/sigmatch.nim

```diff
@@ -1739,11 +1739,22 @@ proc typeRel(c: var TCandidate, f, aOrig: PType,
     let prev = PType(idTableGet(c.bindings, f))
     if prev == nil:
       if aOrig.kind == tyStatic:
-        if f.base.kind != tyNone:
+        if f.base.kind notin {tyNone, tyGenericParam}:
           result = typeRel(c, f.base, a, flags)
           if result != isNone and f.n != nil:
             if not exprStructuralEquivalent(f.n, aOrig.n):
               result = isNone
+        elif f.base.kind == tyGenericParam:
+          # Handling things like `type A[T; Y: static T] = object`
```

Comment:
These things should have been taken care of by the line: 

  result = typeRel(c, f.base, a, flags)

Already. This solution of yours is very good in that it improves the current situation, but it's an ad-hoc bandaid.

Reply:
The information is lost in the future steps so eventually it gets `T` vs. `T` which is ok to solve with a change to the `tyTypedesc` branch, but it then gets `typedesc[int]` vs. `int literal (2)` without knowing either side is static. Seems the only place to solve it is inside the `tyStatic` anywhere lower just lacks the knowledge to know to check the children types. Solving the `typedesc[int]` vs. `int literal` just causes a bunch of issues with other generics for obvious reasons that values equate types.